### DISCUSSION TITLE
Visual overview of keyboard shortcuts

### DISCUSSION THREAD

**Discussion by lannonbr** [johnlindquist/kit#160] [Ideas]
It would be interesting to have a view where you could see all of they keyboard shortcuts you have mapped on an on-screen keyboard. I'm thinking something similar to how Adobe Premiere Pro does it (but probably a lot less complex than Premiere):

![image](https://user-images.githubusercontent.com/3685876/115262431-8c40de00-a102-11eb-8823-c25013fe879e.png)

**Comment by lannonbr** [comment]
**Edit (Dec 20, 2021): Updated to use `getScripts()`**

As a starting point, I wrote a script to do this which will list the script alongside the keyboard shortcut:

![Screen Shot 2021-05-01 at 11 16 43 AM](https://user-images.githubusercontent.com/3685876/116786844-f0eb2980-aa6e-11eb-8a71-411f6c0bc1e0.png)

```js
// Menu: List Keyboard Shortcuts
// Description: List out all scripts with keyboard shortcuts
// Shortcut: opt s
// Author: Benjamin Lannon
// Twitter: @lannonbr

let scripts = await getScripts();

let opts = [];

for (let script of scripts) {
  if (script.shortcut) {
    opts.push({
      name: `${script.value}: ${script.shortcut}`,
    });
  }
}

await arg("Current shortcuts", opts);

```

**Reply by johnlindquist** [reply]
The app keeps a `menu-cache.json` in `~/.kenv/cache/menu-cache.json` of all the script metadata

```js
let { menu } = await cli("fns")

inspect(await menu())
```

That will give you all the metadata of every script for building stuff like this. (I may rename that fn to `metadata` in the future 😅)

**Reply by lannonbr** [reply]
That make this script much much cleaner. Thanks @johnlindquist! Updated the script above with the new logic

**Reply by johnlindquist** [reply]
In the spirit of code golf 🏌️‍♂️

```js
await arg(
  "Shortcuts:",
  (await (await cli("fns")).menu())
    .filter(s => s.shortcut)
    .map(s => s.name)
)
```

**Reply by johnlindquist** [reply]
As a note for anyone who lands on this, `await getScripts()` should now return all the scripts + their metadata now:


[Open list-shortcuts in Script Kit](https://scriptkit.com/api/new?name=list-shortcuts&url=https://gist.githubusercontent.com/johnlindquist/4142b7a6fc0e7bfd5d3302e4feaab796/raw/04fa5c0e15eb8dbf18d1c01fd9f4490c3e671400/list-shortcuts.ts")

```js
// Name: List Shortcuts

import "@johnlindquist/kit"

let scripts = await getScripts()

await arg("Shortcuts:", scripts.filter(script => script.shortcut).map(script => {
    return {
        name: script.name,
        description: script.shortcut
    }
}))
```


`menu()` was changed to a function to update the menu/tray icon.
